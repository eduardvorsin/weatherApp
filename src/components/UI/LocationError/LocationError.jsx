import PropTypes from 'prop-types';
import ErrorInfo from '../ErrorInfo/ErrorInfo';

export default function LocationError({
  testId
}) {
  const noAccessLocationIcon = (
    <svg
      viewBox="0 0 50 50"
      fill="none"
    >
      <path strokeLinejoin="round" strokeMiterlimit="10" strokeWidth="1.765" stroke="currentColor" d="M40 17c0 13.604-15 31-15 31S10 30.928 10 17c0-8.285 6.715-15 15-15 8.284 0 15 6.715 15 15Z" />
      <path strokeLinejoin="round" strokeMiterlimit="10" strokeWidth="1.765" stroke="currentColor" d="M25 24a6 6 0 1 0 0-12 6 6 0 0 0 0 12Z" />
      <path strokeLinecap="round" strokeWidth="1.76" stroke="currentColor" d="M10 2.5 40 48" />
    </svg>
  );

  return (
    <ErrorInfo
      testId={testId}
      headingLevel={2}
      isCentered
      icon={noAccessLocationIcon}
      message="Нет доступа к вашему местоположению, попробуйте предоставить доступ и перезагрузить страницу"
    />
  );
}

LocationError.defaultProps = {
  testId: '',
};

LocationError.propTypes = {
  testId: PropTypes.string,
};
