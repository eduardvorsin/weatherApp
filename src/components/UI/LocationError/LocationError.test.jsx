import { render, screen } from '@testing-library/react';
import LocationError from './LocationError';

describe('LocationError component tests', () => {
  it('renders correctly', () => {
    render(<LocationError />);

    expect(screen.getByText(/нет доступа к вашему местоположению/i)).toBeInTheDocument();
  });

  it('default snapshot of component', () => {
    render(<LocationError
      testId="location-error"
    />);

    expect(screen.getByTestId('location-error')).toMatchSnapshot();
  });
});
