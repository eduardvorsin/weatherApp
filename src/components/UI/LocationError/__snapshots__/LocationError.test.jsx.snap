// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationError component tests default snapshot of component 1`] = `
<div
  class="error-info  error-info--center"
  data-testid="location-error"
  role="alert"
>
  <div
    class="error-info__icon-wrapper"
  >
    <svg
      fill="none"
      viewBox="0 0 50 50"
    >
      <path
        d="M40 17c0 13.604-15 31-15 31S10 30.928 10 17c0-8.285 6.715-15 15-15 8.284 0 15 6.715 15 15Z"
        stroke="currentColor"
        stroke-linejoin="round"
        stroke-miterlimit="10"
        stroke-width="1.765"
      />
      <path
        d="M25 24a6 6 0 1 0 0-12 6 6 0 0 0 0 12Z"
        stroke="currentColor"
        stroke-linejoin="round"
        stroke-miterlimit="10"
        stroke-width="1.765"
      />
      <path
        d="M10 2.5 40 48"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="1.76"
      />
    </svg>
  </div>
  <h2
    class="error-info__description"
  >
    Нет доступа к вашему местоположению, попробуйте предоставить доступ и перезагрузить страницу
  </h2>
</div>
`;
